moveit_planners/ompl/ompl_interface/test/load_test_robot.h

```diff
@@ -0,0 +1,130 @@
+
+/* Author: Jeroen De Maeyer */
+
+#pragma once
+
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/robot_state/robot_state.h>
+#include <moveit/utils/robot_model_test_utils.h>
+
+namespace ompl_interface_testing
+{
+/** \brief Robot indepentent test class setup
+ *
+ * The desired robot tests can be impelmented in a derived class in a generic way.
+ *
+ * It is implemented this way to avoid the ros specific test framework
+ * outside moveit_ros.
+ *
+ * (This is an (uglier) alternative to using the rostest framework
+ * and reading the robot settings from the parameter server.
+ * Then we have several rostest launch files that load the parameters
+ * for a specific robot and run the same compiled tests for all robots.)
+ *
+ * based on
+ * https://stackoverflow.com/questions/38207346/specify-constructor-arguments-for-a-google-test-fixture/38218657
+ * (answer by PiotrNycz)
+ *
+ *   --- example.cpp ---
+ *
+ *   #include <gtest/gtest.h>
+ *   #include "load_test_robot.h"
+ *
+ *   class GenericTests : public ompl_interface_testing::LoadTestRobot, public testing::Test
+ *   {
+ *   public:
+ *     GenericTests(const std::string& robot_name, const std::string& group_name)
+ *        : LoadTestRobot(robot_name, group_name) { }
+ *     void SetUp() override { }
+ *     void TearDown() override  { }
+ *
+ *     void someTest()
+ *     {
+ *       // use robot_model_, robot_state_, .. here
+ *       EXPECT_TRUE(true);
+ *     }
+ *   };
+ *
+ *   // now you can run `someTest()` on different robots:
+ *
+ *   class PandaTest : public GenericTests
+ *   {
+ *   protected:
+ *     PandaTest() : GenericTests("panda", "panda_arm") { }
+ *   };
+ *
+ *   TEST_F(PandaTest, someTest)
+ *   {
+ *     someTest();
+ *   }
+ *
+ *   --- end example.cpp ---
+ *
+ *
+ * */
+class LoadTestRobot
+{
+protected:
+  LoadTestRobot(const std::string& robot_name, const std::string& group_name)
+    : group_name_(group_name), robot_name_(robot_name)
+  {
+    // load robot
+    robot_model_ = moveit::core::loadTestingRobotModel(robot_name_);
+    robot_state_ = std::make_shared<robot_state::RobotState>(robot_model_);
+    robot_state_->setToDefaultValues();
+    joint_model_group_ = robot_state_->getJointModelGroup(group_name_);
+
+    // extract useful parameters for tests
+    num_dofs_ = joint_model_group_->getVariableCount();
+    ee_link_name_ = joint_model_group_->getLinkModelNames().back();
+    base_link_name_ = robot_model_->getRootLinkName();
+
+    ROS_INFO_STREAM("Created test robot named: " << robot_name_ << " for planning group " << group_name_);
+  }
+
+protected:
+  const std::string group_name_;
+  const std::string robot_name_;
+
+  moveit::core::RobotModelPtr robot_model_;
+  robot_state::RobotStatePtr robot_state_;
+  const robot_state::JointModelGroup* joint_model_group_;
+
+  std::size_t num_dofs_;
+  std::string base_link_name_;
+  std::string ee_link_name_;
+};
+}  // namespace ompl_interface_testing
```

Comment:
[nit] newline at end of file, you should be able to configure your editor to do this for you when you save a file (I know sublime has this feature)